### PR TITLE
Issue #13086: InnerAssignmentCheck failed for one line code in Java 1…

### PR DESCRIPTION
**Pull request by aayushRedHat**
Related: https://github.com/checkstyle/checkstyle/issues/13086
Diff Regression config: https://gist.githubusercontent.com/aayushRedHat/7197864ed054dca9580c0cc8c441c1fd/raw/ae6d36be203f1228e0dbb013941c8a29ce6427b6/gistfile1.txt

### CODE REVIEW

**Comment by aayushRedHat** [comment]
Github, generate report

**Comment by nrmancuso** [comment]
Let's get hanging PRs across the finish line before we begin review here: https://github.com/checkstyle/checkstyle/pulls/aayushRedhat

**Comment by aayushRedHat** [comment]
> Let's get hanging PRs across the finish line before we begin review here: https://github.com/checkstyle/checkstyle/pulls/aayushRedhat

Please review @nrmancuso @romani

**Review by romani — APPROVED** [review]
ok to merge, report is good

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java [hunk 1]
```diff
@@ -101,6 +101,7 @@ public class InnerAssignmentCheck
             TokenTypes.RESOURCE_SPECIFICATION,
         },
         {TokenTypes.EXPR, TokenTypes.LAMBDA},
+        {TokenTypes.EXPR, TokenTypes.SWITCH_RULE},
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/InnerAssignmentCheck.java:104]: I don't see it being asked anywhere, but why are we damaging lambda expression by removing them from this list?

https://github.com/checkstyle/checkstyle/commit/f798a441fcd58c890ca44b0cf276bca5752b10e2
https://github.com/checkstyle/checkstyle/issues/377

**Review by rnveach — CHANGES_REQUESTED** [review]

**Review by rnveach — APPROVED** [review]
Nevermind, I thought this was a swap, but  I see it is an addition.

As a note, no test case was ever added for https://github.com/checkstyle/checkstyle/issues/377 .

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignmentWithEnhancedSwitch.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
+InnerAssignment
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.innerassignment;
+
+//non-compiled with javac: Compilable with Java14
+public class InputInnerAssignmentWithEnhancedSwitch {
```
Comment (nrmancuso) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignmentWithEnhancedSwitch.java:8]: Please add more test cases:
```
    void m1(final String operation) {
        boolean innerFlag;
        innerFlag = switch (operation) {
            case "Y" -> flag = true; // expected violation
            case "N" -> {
                yield flag = false; // expected violation
            }
            default -> throw new UnsupportedOperationException();
        };
    }

    void m2(final String operation) {
        boolean innerFlag;
        switch (operation) {
            case "Y": flag = true; break; // should be ok
            case "N": {
                flag = false; // should be ok
                break;
            }
            default: throw new UnsupportedOperationException();
        }
    }
```

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Item:

**Comment by nrmancuso** [comment]
@aayushRedHat ping

**Comment by aayushRedHat** [comment]
> @aayushRedHat ping

Sorry @nrmancuso @romani for the delay in my PR. I was on Holidays, will start working tomorrow.

**Comment by nrmancuso** [comment]
We lost contact with PR author, anyone is welcome to pick this up and reuse the code here

**Pull request closed without merging**